packages/teams-js/src/public/video.ts

```diff
@@ -154,7 +170,19 @@ export namespace video {
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
-    registerHandler('video.effectParameterChange', callback, false);
+
+    const callbackHandler = (effectId: string | undefined): void => {
+      Promise.resolve(callback(effectId))
```

Comment:
There are two options for backward compatibility (for those video apps which don't return a promise)

1. Treat them as a normal case and send `videoEffectReadiness/true` back to the client.
2. Treat them as a special case and send `videoEffectReadiness/false` back to the client with a specific reason.

I personally prefer #2 because it makes the situation explicit and leave a room for client to decide on whether the case should be treated as normal or not.

Reply:
After the offline discussion, I will go with sending `videoEffectReadiness/true + "additional info"` back to the client, so that the client doesn't need to treat them as a special failure case while having a chance to identify if it is from an old video app which returns the callback directly which may cause perf alerts.